File: app/src/main/java/com/mithilakshar/mithilapanchang/kathadesc.java
package com.mithilakshar.mithilapanchang;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import android.content.Intent;
import android.content.res.ColorStateList;
import android.content.res.Configuration;
import android.graphics.Color;
import android.os.Bundle;

import android.os.Handler;
import android.speech.tts.TextToSpeech;
import android.speech.tts.TextToSpeechService;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.squareup.picasso.Picasso;

import java.util.Locale;


public class kathadesc extends AppCompatActivity implements TextToSpeech.OnInitListener {

     TextView kathaDesc,kathaTitle;
    ImageView kathaImg;
    private TextToSpeech textToSpeech;

    String kathaD;
    private Handler handler = new Handler();
    private boolean isFabClicked = false;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_kathadesc);

        kathaTitle=findViewById(R.id.kathatitle);
        kathaDesc=findViewById(R.id.kathaDesc);
        kathaImg=findViewById(R.id.kathaImg);
        FloatingActionButton fab = findViewById(R.id.fab);

        Intent intent = getIntent();

        Locale locale=new Locale("hi", "IN");
        Locale.setDefault(locale);

      



        String kathaT = intent.getStringExtra("kathaName");
         kathaD = intent.getStringExtra("kathaStory");
        String kathaI = intent.getStringExtra("kathaUrl");

        kathaTitle.setText(kathaT);
        kathaDesc.setText(kathaD);
        Picasso.get().load(kathaI).into(kathaImg);

        textToSpeech = new TextToSpeech(this, this);
        switchFabColor(fab);





        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                isFabClicked = !isFabClicked;

                if (isFabClicked) {
                    fab.setImageResource(R.drawable.speaker);
                    switchFabColor(fab);



                    delayedTask(500);;
                } else {
                    fab.setImageResource(R.drawable.mutespeaker);
                    switchFabColor(fab);

                    textToSpeech.stop();
                }





            }
        });










    }

    private void switchFabColor(FloatingActionButton fab) {
        if (isFabClicked) {
            // Set the original color if it's switched
            fab.setBackgroundTintList(ContextCompat.getColorStateList(this, R.color.fabColorOriginal));
        } else {
            // Set the switched color
            fab.setBackgroundTintList(ContextCompat.getColorStateList(this, R.color.fabColorSwitched));
        }

    }


    @Override
    public void onInit(int i) {

        if (i == TextToSpeech.SUCCESS) {




                        // Set language
            textToSpeech.setLanguage(Locale.forLanguageTag("hi"));

            // Speak text

            textToSpeech.setPitch(1f);
            textToSpeech.setSpeechRate(0.6f);

        } else {
            // Handle error

        }

    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        // Shutdown TextToSpeech engine
        if (textToSpeech != null) {
            textToSpeech.stop();
            textToSpeech.shutdown();
        }
    }

    protected void onPause() {
        super.onPause();

        if (textToSpeech != null) {
            textToSpeech.stop();
            textToSpeech.shutdown();
        }
    }

    private void delayedTask(int delayMillis) {
        handler.postDelayed(new Runnable() {
            @Override
            public void run() {


                // Your code to be executed after the delay
                textToSpeech.setLanguage(Locale.forLanguageTag("hi"));

                // Speak text
                textToSpeech.setPitch(1f);
                textToSpeech.setSpeechRate(0.6f);
                textToSpeech.speak(kathaD, TextToSpeech.QUEUE_FLUSH, null, null);

            }
        }, delayMillis);
    }
}

File: app/src/main/java/com/mithilakshar/mithilapanchang/kathadesc.kt
package com.mithilakshar.mithilapanchang

import android.os.Bundle
import android.os.Handler
import android.speech.tts.TextToSpeech
import android.speech.tts.TextToSpeech.OnInitListener
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import com.google.android.material.floatingactionbutton.FloatingActionButton
import com.squareup.picasso.Picasso
import java.util.Locale

class kathadesc : AppCompatActivity(), OnInitListener {
    var kathaDesc: TextView? = null
    var kathaTitle: TextView? = null
    var kathaImg: ImageView? = null
    private var textToSpeech: TextToSpeech? = null
    var kathaD: String? = null
    private val handler = Handler()
    private var isFabClicked = false
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_kathadesc)
        kathaTitle = findViewById(R.id.kathatitle)
        kathaDesc = findViewById(R.id.kathaDesc)
        kathaImg = findViewById(R.id.kathaImg)
        val fab = findViewById<FloatingActionButton>(R.id.fab)
        val intent = intent
        val locale = Locale("hi", "IN")
        Locale.setDefault(locale)
        val kathaT = intent.getStringExtra("kathaName")
        kathaD = intent.getStringExtra("kathaStory")
        val kathaI = intent.getStringExtra("kathaUrl")
        kathaTitle.setText(kathaT)
        kathaDesc.setText(kathaD)
        Picasso.get().load(kathaI).into(kathaImg)
        textToSpeech = TextToSpeech(this, this)
        switchFabColor(fab)
        fab.setOnClickListener {
            isFabClicked = !isFabClicked
            if (isFabClicked) {
                fab.setImageResource(R.drawable.speaker)
                switchFabColor(fab)
                delayedTask(500)
            } else {
                fab.setImageResource(R.drawable.mutespeaker)
                switchFabColor(fab)
                textToSpeech!!.stop()
            }
        }
    }

    private fun switchFabColor(fab: FloatingActionButton) {
        if (isFabClicked) {
            // Set the original color if it's switched
            fab.backgroundTintList = ContextCompat.getColorStateList(this, R.color.fabColorOriginal)
        } else {
            // Set the switched color
            fab.backgroundTintList = ContextCompat.getColorStateList(this, R.color.fabColorSwitched)
        }
    }

    override fun onInit(i: Int) {
        if (i == TextToSpeech.SUCCESS) {


            // Set language
            textToSpeech!!.setLanguage(Locale.forLanguageTag("hi"))

            // Speak text
            textToSpeech!!.setPitch(1f)
            textToSpeech!!.setSpeechRate(0.6f)
        } else {
            // Handle error
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        // Shutdown TextToSpeech engine
        if (textToSpeech != null) {
            textToSpeech!!.stop()
            textToSpeech!!.shutdown()
        }
    }

    override fun onPause() {
        super.onPause()
        if (textToSpeech != null) {
            textToSpeech!!.stop()
            textToSpeech!!.shutdown()
        }
    }

    private fun delayedTask(delayMillis: Int) {
        handler.postDelayed({ // Your code to be executed after the delay
            textToSpeech!!.setLanguage(Locale.forLanguageTag("hi"))

            // Speak text
            textToSpeech!!.setPitch(1f)
            textToSpeech!!.setSpeechRate(0.5f)
            val chunkSize = 4000
            var i = 0
            while (i < kathaD!!.length) {
                val endIndex = Math.min(i + chunkSize, kathaD!!.length)
                val chunk = kathaD!!.substring(i, endIndex)
                textToSpeech!!.speak(chunk, TextToSpeech.QUEUE_ADD, null, null)
                i += chunkSize
            }
        }, delayMillis.toLong())
    }
}